sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/parser/AstBuilder.scala

```diff
@@ -914,19 +914,19 @@ class AstBuilder extends SqlBaseBaseVisitor[AnyRef] with SQLConfHelper with Logg
       selectExpressions: Seq[NamedExpression],
       query: LogicalPlan): LogicalPlan = withOrigin(ctx) {
     if (ctx.groupingExpressionsWithGroupingAnalytics.isEmpty) {
-      val groupByExpressions = expressionList(ctx.groupingExpressions)
       if (ctx.GROUPING != null) {
-        // GROUP BY .... GROUPING SETS (...)
-        // `GROUP BY warehouse, product GROUPING SETS((warehouse, producets), (warehouse))` is
-        // semantically equivalent to `GROUP BY GROUPING SETS((warehouse, produce), (warehouse))`.
-        // Under this grammar, the fields appearing in `GROUPING SETS`'s groupingSets must be a
-        // subset of the columns appearing in group by expression.
+        // GROUP BY GROUPING SETS (...)
+        if (!ctx.groupingExpressions.isEmpty) {
+          logWarning("The groupBy expressions between 'GROUP BY' and 'GROUPING SETS' will " +
+            "be ignored and this syntax is deprecated. Please use 'GROUP BY GROUPING SETS (...)' " +
+            "directly.")
+        }
```

Comment:
One waring is  the behavior of 
```
GROUP BY a, b, c GROUPING SETS((a, b), (a), (b))
```
will change after this.

Reply:
how is it changed?

Reply:
Re check the code, not a  problem.